service/s3control/internal/customizations/process_outpost_id.go

```diff
@@ -67,8 +67,14 @@ func (m *processOutpostIDMiddleware) HandleSerialize(
 		return out, metadata, fmt.Errorf("dualstack is not supported for outposts request")
 	}
 
-	// set request url
-	req.URL.Host = serviceEndpointLabel + requestRegion + ".amazonaws.com"
+	// if endpoint source is a custom endpoint source, do not modify host prefix.
+	// This is a requirement for s3 vpc interface endpoints.
+	if v := awsmiddleware.GetEndpointSource(ctx); v != aws.EndpointSourceCustom {
+		serviceEndpointLabel := "s3-outposts."
+
+		// set request url
+		req.URL.Host = serviceEndpointLabel + requestRegion + ".amazonaws.com"
+	}
```

Comment:
is `requestRegion` the region derived from an endpoint resolve, or the region the user configured the client with. If this is the region the user configured the client with this coudl be a pseudo region, e.g. `us-west-2-fips` which would fail here.

Reply:
Request region is the resolved region. It may come from ARN or client config . 

We have validations in-place to assert region is not a fips region.

```
	// validate if fips
	if s3shared.IsFIPS(requestRegion) {
		return out, metadata, fmt.Errorf("unsupported fips region provided for outposts request")
```